### PR TITLE
Fix code splitting

### PR DESCRIPTION
**Pull request by baldurh**
This PR adds necessary changes so that code splitting works properly

### CODE REVIEW

File: config/extenders/codeSplitting.js [hunk 1]
```diff
@@ -72,18 +73,40 @@ export default (webpackConfig, buildOptions) => {
...
+      ifClient(new webpack.HashedModuleIdsPlugin()),
+
+      // Add vendor code chunk
+      ifProdClient(
         new webpack.optimize.CommonsChunkPlugin({
-          names: ['bootstrap'], // needed to put webpack bootstrap code before chunks
+          name: 'vendor',
           filename: isDev ? '[name].js' : '[name]-[chunkhash].js',
-          minChunks: Infinity,
+          minChunks: module => /node_modules/.test(module.resource),
```
Comment (osk) [inline config/extenders/codeSplitting.js]: What does this do?

File: config/values.js [hunk 2]
```diff
@@ -11,6 +11,35 @@ import codeSplittingConfigExtender from './extenders/codeSplitting';
...
+function getPublicPath() {
+  if (NODE_ENV === 'development') {
+    // As we run a seperate development server for our client and server
+    // bundles we need to use an absolute http path for the public path.
+    return `${getPublicUrl()}${clientBundleWebPath}`;
+  }
+  // Otherwise we expect our bundled client to be served from this path.
+  return clientBundleWebPath;
+}
+
```
Comment (osk) [inline config/values.js]: I think this should be moved into its own file, a lot of stuff you don't want to look at day to day

**Review by osk — COMMENTED** [review]
Looks good! Although a bit (a lot) of it is like black magic

File: config/extenders/codeSplitting.js (hunk 1 above)
Comment (baldurh) [reply config/extenders/codeSplitting.js]: module.context && module.context.includes("node_modules");

Comment (baldurh) [reply config/extenders/codeSplitting.js]: This adds every module imported from `node_modules` into a single chunk. Will add a comment and make it a little bit clearer.

File: config/values.js (hunk 2 above)
Comment (baldurh) [reply config/values.js]: Agreed. Will move.

File: config/extenders/codeSplitting.js [hunk 3]
```diff
@@ -72,18 +73,42 @@ export default (webpackConfig, buildOptions) => {
...
+      // Add vendor code chunk
+      ifProdClient(
         new webpack.optimize.CommonsChunkPlugin({
-          names: ['bootstrap'], // needed to put webpack bootstrap code before chunks
+          name: 'vendor',
           filename: isDev ? '[name].js' : '[name]-[chunkhash].js',
-          minChunks: Infinity,
+          // Put all node_modules into one chunk
+          // see: https://webpack.js.org/plugins/commons-chunk-plugin/#passing-the-minchunks-property-a-function
+          minChunks: module => module.context && module.context.includes('node_modules'),
```
Comment (osk) [inline config/extenders/codeSplitting.js:99]: 🔥

File: config/values.js [hunk 4]
```diff
@@ -6,11 +6,14 @@
  */
 
 import * as EnvVars from './utils/envVars';
+import { getPublicUrl, getPublicPath } from './utils/publicPath';
```
Comment (osk) [inline config/values.js:9]: 😍

**Pull request merged**